### PR TITLE
Document importing components with @using directives

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/11829

### CODE REVIEW

**Comment by ajaybhargavb** [comment]
@guardrex, the preview doesn't show the `[!NOTE]` section correctly. Is that expected?

File: aspnetcore/razor-components/components.md [hunk 1]
```diff
@@ -479,6 +479,32 @@ The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/razor-com
 
 The base class should derive from `ComponentBase`.
 
+## Importing components
+
+Components infer their namespace based on the file path and the file name.
```
Comment (rynowak) [inline aspnetcore/razor-components/components.md]: I would saw that the namespace of a component authored with Razor is based on:
- the project's `RootNamespace`
- the path from the project root to the component

That's a little more accurate.

File: aspnetcore/razor-components/components.md [hunk 2]
```diff
@@ -479,6 +479,32 @@ The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/razor-com
...
+Example,
+`ComponentsSample/Pages/Index.razor` will be in the namespace `ComponentsSample.Pages`. So, all components in the same folder `Pages` and the parent folder `ComponentsSample` are already in scope.
```
Comment (rynowak) [inline aspnetcore/razor-components/components.md]: I wouldn't start with this example. I would say that the bindings between tag names and component classes are based on usings. When you explain these concepts in terms of normal C# name binding rules they make much more sense.

**Comment by guardrex** [comment]
@ajaybhargavb ... I'll look :eyes: right after the engineering lingo is worked out. I'll perform a quick grammar+style pass on it.

**Comment by guardrex** [comment]
@ajaybhargavb I think we'll need to reconstitute this PR. These topics (*this one*) will be physically moving on the docs repo on Monday morning (time frame of ... say 🤔 ... noon CST, which is 10am PST), and that's going to :boom: this PR.

Would you like to reconstitute it, or would you like me to take care of it?

**Comment by ajaybhargavb** [comment]
@guardrex, sorry I missed your edit. I'll reconstitute this PR with the feedback addressed and then you can take it from there. Does that sound good?

**Comment by guardrex** [comment]
@ajaybhargavb Yes ... I'm mere minutes away from submitting a commit that will radically change the physical file locations. It will break this PR. 

There are just a couple of sections here to paste into a new PR, so it's an easy job to just submit a new PR.

I'll ping u here when we get the updates merged to master. Then, you can close this and set up a new PR on the new location for this topic.

**Comment by ajaybhargavb** [comment]
Sounds good. Thanks @guardrex

**Comment by guardrex** [comment]
@ajaybhargavb Done ... https://github.com/aspnet/AspNetCore.Docs/pull/11984.

Reconstitute the PR on https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/components.md.

**Comment by guardrex** [comment]
@ajaybhargavb I can do it if ur busy.

**Comment by ajaybhargavb** [comment]
@guardrex, please go ahead if you have the bandwidth. Otherwise, I'll get to this later today.

**Comment by guardrex** [comment]
Sure ... I'm sitt'in and waiting on "developments" on the engineering side. I'll take care of it and ping u on the PR.

**Pull request closed without merging**